### PR TITLE
Review/small test avoid validation warnings

### PR DESCRIPTION
**Pull request by infapi00**

### CODE REVIEW

File: framework/vkdf-init.cpp [hunk 1]
```diff
@@ -308,8 +308,10 @@ choose_device_extensions(VkdfContext *ctx,
          else
             vkdf_info("Optional extension '%s' not available.\n", ext);
       } else {
-         ctx->device_extensions.enabled[i] = true;
-         (*enabled_extensions)[(*enabled_extension_count)++] = ext;
+	 if (extensions[i].required) {
+	    ctx->device_extensions.enabled[i] = true;
+	    (*enabled_extensions)[(*enabled_extension_count)++] = ext;
+	 }
       }
```
Comment (itoral) [inline framework/vkdf-init.cpp:316]: Indentation looks off here.

File: framework/vkdf-init.cpp [hunk 2]
```diff
@@ -64,13 +70,21 @@ get_required_extensions(VkdfContext *ctx, bool enable_validation)
...
+      ctx->inst_extensions[i++] =
          g_strdup(VK_EXT_DEBUG_REPORT_EXTENSION_NAME);
    }
+
+   if (!ctx->no_swapchain) {
+      ctx->inst_extensions[i++] =
+         g_strdup(VK_KHR_SURFACE_EXTENSION_NAME);
+   }
+
+   for (uint32_t c = 0; c < platform_extension_count; c++) {
```
Comment (itoral) [inline framework/vkdf-init.cpp:85]: let's assert that platform_extension_count + i < ctx->inst_extension_count before this loop

Comment (infapi00) [reply framework/vkdf-init.cpp:85]: added but with <=

File: framework/vkdf-init.cpp (hunk 1 above)
Comment (infapi00) [reply framework/vkdf-init.cpp:316]: Yep, it seems that I added some tabs

**Pull request closed without merging**